From e70b5ff0db2b5975dcb2b1c7024a2d55027f824b Mon Sep 17 00:00:00 2001
From: Seungha Yang <seungha@centricular.com>
Date: Tue, 14 May 2024 23:04:59 +0900
Subject: [PATCH] pangowin32: Fix PangoCoverage and hb_set_t leaks

Don't need to increase refcount of just created PangoCoverage created,
and release hb_set_t after use.
---
 pango/pangowin32.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/pango/pangowin32.c b/pango/pangowin32.c
index 6781e179..67b8675c 100644
--- a/pango/pangowin32.c
+++ b/pango/pangowin32.c
@@ -928,7 +928,8 @@ pango_win32_font_get_coverage (PangoFont     *font,
       while (hb_set_next(chars, &ch))
         pango_coverage_set (coverage, ch, PANGO_COVERAGE_EXACT);
 
-      win32face->coverage = g_object_ref (coverage);
+      hb_set_destroy (chars);
+      win32face->coverage = coverage;
     }
 
   return g_object_ref (win32face->coverage);
-- 
2.42.0.windows.2

